drivers/gpio/gpio_rpi_pico.c

```diff
@@ -50,11 +50,9 @@ static int gpio_rpi_configure(const struct device *dev,
 		}
 	} else if (flags & GPIO_INPUT) {
 		gpio_set_dir(pin, GPIO_IN);
-		if (flags & GPIO_PULL_UP) {
-			gpio_pull_up(pin);
-		} else if (flags & GPIO_PULL_DOWN) {
-			gpio_pull_down(pin);
-		}
+		gpio_set_pulls(pin,
+			!!(flags & GPIO_PULL_UP),
```

Comment:
The `!!` is not necessary as `gpio_set_pulls` does it itself. See 

https://github.com/raspberrypi/pico-sdk/blob/2062372d203b372849d573f252cf7c6dc2800c0a/src/rp2_common/hardware_gpio/gpio.c#L55
and
https://github.com/raspberrypi/pico-sdk/blob/2062372d203b372849d573f252cf7c6dc2800c0a/src/common/pico_base/include/pico/types.h#L91

Reply:
I dunno, the function declaration is still `bool`, so it seems like we should be passing the right thing, even if the current implementation allows us not to.